lib/import/brca/providers/st_george/st_george_handler.rb

```diff
@@ -1,436 +1,360 @@
-            elsif targeted_test?(record)
-              genotype.add_test_scope(:targeted_mutation)
-            elsif full_screen?(record)
-              genotype.add_test_scope(:full_screen)
-            elsif void_genetictestscope?(record)
-              @logger.debug 'Unknown moleculartestingtype'
-              genotype.add_test_scope(:no_genetictestscope)
+            genotypes.each do |single_genotype|
+              process_variants(single_genotype, record)
+              @persister.integrate_and_store(single_genotype)
             end
           end
 
-          def process_variants_from_record(genotype, record)
+          def fill_genotypes(genotype, record)
             genotypes = []
-            positive_gene = get_positive_genes(record)
-            if ashkenazi?(record) || polish?(record) || full_screen?(record)
-              process_fullscreen_records(genotype, record, positive_gene, genotypes)
-            elsif targeted_test?(record) || void_genetictestscope?(record)
-              process_targeted_records(positive_gene, genotype, record, genotypes)
-            end
-            genotypes
-          end
-
-          def get_positive_genes(record)
-            positive_gene = []
-            gene = record.raw_fields['genotype'].scan(BRCA_GENES_REGEX)
-            deprecated_gene = record.raw_fields['genotype'].scan(DEPRECATED_BRCA_NAMES_REGEX)
-            process_rightname_gene(gene, positive_gene) if gene.present?
-            process_deprecated_gene(deprecated_gene, positive_gene) if deprecated_gene.present?
-            @logger.debug 'Unable to extract gene' if gene.empty? && deprecated_gene.empty?
-            positive_gene
-          end
-
-          def process_rightname_gene(gene, positive_genes)
-            gene.size == 1 ? positive_genes.append(gene.join) : positive_genes.append(gene)
-          end
-
-          def process_deprecated_gene(deprecated_gene, positive_genes)
-            if deprecated_gene.size == 1
-              positive_genes.append(DEPRECATED_BRCA_NAMES_MAP[deprecated_gene.join])
-            else
-              deprecated_gene.each do |dg|
-                positive_genes.append(DEPRECATED_BRCA_NAMES_MAP[dg])
-              end
-            end
-          end
-
-          def process_fullscreen_records(genotype, record, positive_genes, genotypes)
-            if normal?(record)
-              normal_full_screen(genotype, genotypes)
-            elsif failed_test?(record)
-              failed_full_screen(genotype, genotypes)
-            elsif positive_cdna?(record) || positive_exonvariant?(record)
-              if record.raw_fields['genotype'].scan(CDNA_REGEX).size > 1
-                process_multiple_positive_variants(positive_genes, genotype, record, genotypes)
-              else
-                single_variant_full_screen(genotype, genotypes, positive_genes, record)
+            if genotype.attribute_map['genetictestscope'] == 'Targeted BRCA mutation test'
```

Comment:
use base class methods -> genotype.targeted?

Reply:
updated in [b72975c](https://github.com/NHSDigital/data_management_system/pull/77/commits/b72975c9bb7ce4cd2ac1e0fa3bd23ff936a97093)